gto/api.py

```diff
@@ -86,7 +87,26 @@ def annotate(
     # update: bool = False,
 ):
     """Add an artifact to the Index"""
-    return init_index_manager(path=repo).add(
+    if isinstance(path, str) and is_url_of_remote_repo(repo=path):
+        try:
+            with cloned_git_repo(repo=path) as tmp_dir:
+                return init_index_manager(path=repo).add(
+        name,
+        type=type,
+        path=tmp_dir,
+        must_exist=must_exist,
+        labels=labels,
+        description=description,
+        update=True,
+    )
+        except (NotADirectoryError, PermissionError) as e:
+            raise e.__class__(
+                "Are you using windows with python < 3.9? "
+                "This may be the reason of this error: https://bugs.python.org/issue42796. "
+                "Consider upgrading python."
+            ) from e
+    else:
+        return init_index_manager(path=repo).add(
```

Comment:
> Notice how many rather intrusive changes were required. The most annoying part is init_index_manager, which I only started to address.

Yes, I guess here I was unable to properly generalize here, to simplify future changes. Generally, this should be moved to `RepoIndexManager` I guess.